### DISCUSSION TITLE
Extensions for Azure IoT/AWS and IoT PnP

### DISCUSSION THREAD

**Discussion by ridomin** [dotnet/MQTTnet#1512] [Show and tell]
I'm working on some extensions to facilitate:
- Connection to different endpoints with a pre-configured connection/auth options (Azure IoT Hub, AWS IoT Core, Hive MQ, Mosquitto and others.
- Support Basic Auth (username/password) and Client Certs.
- Implement the Conventions to define Telemetry, Properties and Commands, using different topic schemes.

I'm thinking of publishing these packages under the `MQTTnet.Extensions.MultiCloud` namespace.

Are these package names resonable? (not sure if I can use the `MQTTnet.Extensions.*` prefix)

I'm looking to make the code available soon, and happy to add any collaborator who wants to participate or provide feedback.

**Comment by ridomin** [comment]
Since I got no feedback at all, I proceed and published

https://github.com/iotmodels/MQTTnet.Extensions.MultiCloud with corresponding NuGets

**Comment by chkr1011** [comment]
You can use these package names but please make sure to not use any of the nuget icons of the official packages.

**Reply by ridomin** [reply]
Sure thing, I'm using a custom icon.

Would you like to include a link in the docs/wiki?

**Reply by chkr1011** [reply]
Please create a PR with the changes in the Readme section. I will move all references to the wiki in the future.
